pkg/rke2/rke2.go

```diff
@@ -166,5 +176,151 @@ func setup(clx *cli.Context, cfg Config) error {
 	}
 	executor.Set(&sp)
 
+	disabledItems := map[string]bool{
+		"kube-apiserver":          disableAPIServer,
+		"kube-scheduler":          disableScheduler,
+		"kube-controller-manager": disableControllerManager,
+		"etcd":                    disableETCD,
+	}
+	return removeOldPodManifests(dataDir, disabledItems)
+}
+
+func podManifestsDir(dataDir string) string {
+	return filepath.Join(dataDir, "agent", defaultPodManifestPath)
+}
+
+func removeOldPodManifests(dataDir string, disabledItems map[string]bool) error {
+	var kubeletStandAlone bool
+
+	kubeletErr := make(chan error)
+	containerdErr := make(chan error)
+
+	ctx := context.Background()
+
+	for component, disabled := range disabledItems {
+		if disabled {
+			manifestName := filepath.Join(podManifestsDir(dataDir), component+".yaml")
+			if _, err := os.Stat(manifestName); err == nil {
+				kubeletStandAlone = true
+				if err := os.Remove(manifestName); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	if kubeletStandAlone {
+		kubeletCmd := exec.Command("kubelet")
+		containerdCmd := exec.Command("containerd")
+
+		tmpAddress := filepath.Join(os.TempDir(), "containerd.sock")
+
+		// start containerd
+		go startContainerd(ctx, dataDir, containerdErr, tmpAddress, containerdCmd)
+		// start kubelet
+		go startKubelet(ctx, dataDir, kubeletErr, tmpAddress, kubeletCmd)
+
+		// check for any running containers from the disabled items list
+		go checkForRunningContainers(ctx, tmpAddress, disabledItems, kubeletErr, containerdErr)
+
+		go func() {
+			// kill both temp processes after 5 minutes anyway
+			time.Sleep(5 * time.Minute)
```

Comment:
Rather than running a sleeping goroutine to enforce the timeout, consider doing this this via `context.WithDeadline`
https://golang.org/pkg/context/#WithDeadline